### ISSUE TITLE
Use microlens instead of lens

### ISSUE THREAD

**Issue by leptonyu** [haskell-servant/servant-swagger#109]
Lens is very heavy library, many libraries use microlens instead. Can you replace it?

**Comment by phadej** [comment]
I'm sorry, but the answer is *no*.

We (me and @alpmestan) agreed that`lens` dependencies can be added to `servant-*` packages.
`lens` is virtually unavoidable in any non-trivial real world`servant` applications, something in your dependencies transitively depends on it (even you don't use `lens` yourself) or you have to be extra careful when picking them.
E.g. `servant-client` depends on `kan-extensions` since recently.

Also it's `swagger2` which pulls in `lens` and it defines `Prism`s in https://hackage.haskell.org/package/swagger2-2.4/docs/Data-Swagger-Lens.html, so `profunctors` would be needed with `microlens` ecosystem too. 

There aren't many packages between `profunctors` and `lens`. Even less between `kan-extensions` and `lens`.

`cabal`, `stack` and `nix` are all good quite good at caching dependencies, and the acme-kmett is quite stable (new releases happen rarely).

**Comment by leptonyu** [comment]
Thanks. 

It seems `acme-kmett`'s [README.md](https://github.com/phadej/acme-kmett/blob/master/README.md) needs a little modification. 

GHC-8.6.4 -> GHC-8.6.5

**Comment by alpmestan** [comment]
Right, I think it's a bit tricky for _us_ to change this. But if I'm not mistaken, none of servant, servant-client[-core] nor servant-server depend on lens (transitively), do they? Only the -docs/-swagger/... packages do, IIRC.

**Comment by neongreen** [comment]
`microlens` itself is lens-lib-agnostic. As in, you don't lose anything by using microlens instead of lens — van Laarhoven lenses are compatible with each other.

If you are exporting prisms, though, you're better off with `lens`.

**Comment by akhesaCaro** [comment]
Hi,
Servant-swagger will be moved into the main Servant repo (see : https://github.com/haskell-servant/servant/pull/1475)
If this issue is still relevant, would it be possible for you to summit it there? : https://github.com/haskell-servant/servant/issues

Thanks in advance!
